### PR TITLE
sqlcipher support

### PR DESCRIPTION
**Pull request by biohazardffm**
- compile using sqlcipher repository
- add key and rekey functions; rekey(0,0) does not work yet as I could not figure out the right type in swift

### CODE REVIEW

**Comment by stephencelis** [comment]
Nice! Thanks for the PR!

Is there any way to compose this so it's opt-in given a build flag, and without SQLCipher it uses the OS-bundled libraries?

**Comment by biohazardffm** [comment]
Welcome! Not sure but I think it should be possible. Tried to do an extension on Database before in a different framework but that did not work....
Still learning swift... Will try to add a build flag and get back to you.

File: SQLite Common/Database.swift [hunk 1]
```diff
@@ -51,6 +51,25 @@ public final class Database {
...
+	public func key(key:String) {
+		try(sqlite3_key(handle, key, Int32(key.lengthOfBytesUsingEncoding(NSUTF8StringEncoding))))
+	}
+
+	/// Change the key on an open database. If the current database is not
+	/// encrypted, this routine will encrypt it.
+	/// If key == "" the database is decrypted.
+	public func rekey(key:String) {
+		if(key.lengthOfBytesUsingEncoding(NSUTF8StringEncoding) > 0) {
+			try(sqlite3_rekey(handle, key, Int32(key.lengthOfBytesUsingEncoding(NSUTF8StringEncoding))))
```
Comment (stephencelis) [inline SQLite Common/Database.swift]: You should be able to keep the byte length Swift-native by calling:

``` swift
countElements(key.utf8)
```

Comment (biohazardffm) [inline SQLite Common/Database.swift]: works, but still have to wrap it in an Int32()

**Comment by biohazardffm** [comment]
I have added 2 new sqlcipher enabled targets. Will add some tests and then it should be good to go.

**Comment by stephencelis** [comment]
Thanks for your work on this! Merged it and rebased my universal framework branch against it. Should simplify things by only requiring two targets.

**Comment by manojshrestha** [comment]
HI Stephencelis,
I was trying to add SQLCipher in my project but there is error saying "No such module SQLiteCipher". I have added it in both Embed frameworks and Target Dependencies. (Manual Installation)

It would be helpful if you add runnable example project with the code.
Thank you,

**Pull request merged**